tests/RingBuffer.fs

```diff
@@ -0,0 +1,44 @@
+module Elmish.RingBufferTests
+
+open FsCheck.NUnit
+open System.Collections.Generic
+open Swensen.Unquote
+
+type Op =
+    | Pop
+    | Push of int
+
+type OpResult =
+    | Poped of int option
+    | Pushed
+
+let internal applyQOps (ops:Op list) (q:int Queue) =
+    ops
+    |> List.map (function 
+                 | Pop -> 
+                   match q.TryDequeue() with (true, v) -> Some v | _ -> None
+                   |> Poped
+                 | Push v -> 
+                   q.Enqueue v
+                   Pushed)
+
+let internal applyRBOps (ops:Op list) (rb:int RingBuffer) =
+    ops
+    |> List.map (function 
+                 | Pop -> 
+                   Poped <| rb.Pop()
+                 | Push v -> 
+                   rb.Push v
+                   Pushed)
+                   
+[<Property>]
+let actsLikeAQueue (ops:Op list) =  
+    let q = Queue()
+    let rb = RingBuffer 10
+    (q |> applyQOps ops) =! (rb |> applyRBOps ops)
```

Comment:
Here we're asserting that the same operations on the queue and the buffer result in the same outcome.